frontend/src/components/ExplorerCard/ExplorerCard.tsx

```diff
@@ -0,0 +1,329 @@
+	};
+
+	const {
+		stagedQuery,
+		currentQuery,
+		panelType,
+		redirectWithQueryBuilderData,
+	} = useQueryBuilder();
+
+	const {
+		data: viewsData,
+		isLoading,
+		error,
+		isRefetching,
+		refetch: refetchAllView,
+	} = useGetAllViews(sourcepage);
+
+	useErrorNotification(error);
+
+	const handlePopOverClose = (): void => {
+		setIsOpen(false);
+	};
+
+	const handleOpenChange = (newOpen: boolean): void => {
+		setIsOpen(newOpen);
+	};
+
+	const viewName =
+		useGetSearchQueryParam(querySearchParams.viewName) || 'Query Builder';
+
+	const viewKey = useGetSearchQueryParam(querySearchParams.viewKey) || '';
+
+	const { mutateAsync: updateViewAsync } = useUpdateView({
+		compositeQuery: mapCompositeQueryFromQuery(currentQuery, panelType),
+		viewKey,
+		extraData: '',
+		sourcePage: sourcepage,
+		viewName,
+	});
+
+	const { mutateAsync: deleteViewAsync } = useDeleteView(viewKey);
+
+	const onDeleteHandler = useCallback(async () => {
+		deleteViewHandler({
+			deleteViewAsync,
+			notifications,
+			panelType,
+			redirectWithQueryBuilderData,
+			refetchAllView,
+			viewId: viewKey,
+			viewKey,
+		});
+	}, [
+		deleteViewAsync,
+		notifications,
+		panelType,
+		redirectWithQueryBuilderData,
+		refetchAllView,
+		viewKey,
+	]);
+
+	const onUpdateQueryHandler = useCallback(async () => {
+		await updateQueryHandler({
+			compositeQuery: mapCompositeQueryFromQuery(currentQuery, panelType),
+			viewKey,
+			extraData: '',
+			sourcePage: sourcepage,
+			viewName,
+			notifications,
+			setIsQueryUpdated,
+			updateViewAsync,
+		});
+		refetchAllView();
+	}, [
+		currentQuery,
+		notifications,
+		panelType,
+		refetchAllView,
+		sourcepage,
+		updateViewAsync,
+		viewKey,
+		viewName,
+	]);
+
+	const onMenuItemSelectHandler = useCallback(
+		({ key }: { key: string }): void => {
+			const currentViewDetails = getViewDetailsUsingViewKey(
+				key,
+				viewsData?.data?.data,
+			);
+			if (!currentViewDetails) return;
+			const { query, name, uuid } = currentViewDetails;
+
+			// AffregateOperator should be noop for list and trace Panel Type and count for graph and table Panel Type.
+			query.builder.queryData = query.builder.queryData.map((item) => {
+				const newItem = item;
+				if (currentPanelType === 'list' || currentPanelType === 'trace') {
+					newItem.aggregateOperator = 'noop';
+				} else {
+					newItem.aggregateOperator = 'count';
+				}
+				return newItem;
+			});
+
+			redirectWithQueryBuilderData(query, {
+				[queryParamNamesMap.panelTypes]: panelType,
+				[querySearchParams.viewName]: name,
+				[querySearchParams.viewKey]: uuid,
+			});
+		},
+		[
+			viewsData?.data?.data,
+			redirectWithQueryBuilderData,
+			panelType,
+			currentPanelType,
+		],
+	);
+
+	useEffect(() => {
+		if (copyUrl.value) {
+			notifications.success({
+				message: 'Copied to clipboard',
+			});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [copyUrl.value]);
+
+	useEffect(() => {
+		setIsQueryUpdated(
```

Comment:
can't we move this logic to onSuccess useEffect should be avoided in handling async operation

Reply:
I don't think so, as there is `stagedQuery` involved, this should be called every time `stagedQuery` changes.  Also, there is no async operation here.

Reply:
we should not handle this operation via useEffect as unnecessary compute would be going on

Reply:
we could handle this one via onStageQueryUpdate method exposed via QueryBuilder Provider but useEffect is not required